### PR TITLE
fix(typescript): Fix errors from linters and an update for signatures

### PR DESCRIPTION
**Pull request by wolfy1339**
You can't use ESModule and CommonJS exports in the same file (you could theoretically, but it's bad practice), so we have to move the definitions into the namespace.

This does not affect the way to import the types, you can still do this:
```ts
import { WebhookPayloadStatus } from '@octokit/webhooks';
```

In the `Options` type, replace the `T` parameter with `any` since it was not defined.

### CODE REVIEW

File: scripts/generate-types.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ webhooks.forEach(({ name, actions, examples }) => {
     ...actions.map(action => `'${name}.${action}'`)
   ].join(' | ')
   signatures.push(`
-    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<${typeName}>) => void): void
+    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<Webhhooks.${typeName}>) => void): void
```
Comment (gr2m) [inline scripts/generate-types.js]:
```suggestion
    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<Webhooks.${typeName}>) => void): void
```

**Review by gr2m — COMMENTED** [review]
could you run `npm run generate-types` to also update the index.d.ts file?

File: scripts/generate-types.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ webhooks.forEach(({ name, actions, examples }) => {
     ...actions.map(action => `'${name}.${action}'`)
   ].join(' | ')
   signatures.push(`
-    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<${typeName}>) => void): void
+    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<Webhooks.${typeName}>) => void): void
+    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<Webhooks.${typeName}>) => Promise<void>): void
```
Comment (gr2m) [inline scripts/generate-types.js]: I wonder, can you combine the two lines above?

```ts
public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<Webhooks.${typeName}>) => void | Promise<void>): void
```

Comment (wolfy1339) [reply scripts/generate-types.js]: I'm not sure if It could work, I'll go test

**Comment by wolfy1339** [comment]
@gr2m This is ready

**Review by gr2m — APPROVED** [review]
Thanks for the ping!

**Pull request merged**